ovs/extensions/healthcheck/expose_to_cli.py

```diff
@@ -0,0 +1,324 @@
+                                             'module_name': name,
+                                             'function': submember[1].__name__,
+                                             'class': member[1].__name__,
+                                             'location': file_path,
+                                             'version': version_id
+                                             })
+            # Write the dict to a temp file
+            with open(temp_file_path, 'w') as f2:
+                f2.write(str(found_items))
+            return found_items
+
+        try:
+            with open(temp_file_path, 'r') as f:
+                exposed_methods = ast.literal_eval(f.read())
+        except IOError:
+            # If file doesn't exist
+            exposed_methods = None
+
+        result = None
+        # Search first to use old cache
+        if exposed_methods:
+            if not datetime.strptime(exposed_methods['expires'], time_format) > datetime.now() + timedelta(hours=2):
+                result = search_dict(exposed_methods)
+        if not result:
+            exposed_methods = build_cache()
+            result = search_dict(exposed_methods)
+        return result
+
+    @classmethod
+    def print_help(cls, module='', method=''):
+        """
+        Prints the possible methods that are exposed to the CLI
+        :param module: module name specified with the cli
+        :type module: str
+        :param method: method name specified with the cli
+        :type method: str
+        :return: found cache
+        :rtype: dict
+        """
+        print "Possible optional arguments are:"
+        for arg in cls.OPTIONAL_ARGUMENTS:
+            print "ovs healthcheck [module] [method] {0}".format(arg)
+        cache = cls._discover_methods(module, method)
+        if module:
+            if cache is None:
+                print "Found no methods for module {0}".format(module)
+                return cls.print_help()
+            else:
+                print "Possible options for '{0}' are: ".format(module)
+        else:
+            print "Possible options are: "
+        for module in cache:
+            for option in cache[module]:
+                print "ovs healthcheck {0} {1}".format(module, option['method_name'])
+        return cache
+
+
+class HealthCheckCLIRunner(CLIRunner):
+    """
+    Healthcheck adaptation of CLIRunner
+    Injects a result_handler instance with shared resources to every test to collect the results.
+    """
+    logger = LogHandler.get("healthcheck", "clirunner")
+    OPTIONAL_ARGUMENTS = CLIRunner.OPTIONAL_ARGUMENTS + ['--to-json', '--unattended']
+    START_PATH = ''.join([os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir)), '/healthcheck'])
+
+    @staticmethod
+    def run_method(*args):
+        """
+        Executes the given method
+        :return:
+        """
+        module_name = ''
```

Comment:
Big parts of this `run_method` method are similar to the `CLIRunner.run_method`. Thats not proper inheritance